src/server/command_registry.h

```diff
@@ -84,14 +85,41 @@ struct CommandContext {
   ConnectionContext* conn_cntx;
 };
 
+// TODO: move it to helio
+// Makes sure that the POD T that is passed to the constructor is reset to default state
+template <typename T> class MoveOnly {
+ public:
+  MoveOnly() = default;
+
+  MoveOnly(const MoveOnly&) = delete;
+  MoveOnly& operator=(const MoveOnly&) = delete;
+
+  explicit MoveOnly(MoveOnly&& t) noexcept : value_(std::move(t.value_)) {
+    t.value_ = T{};  // Reset the passed value to default state
```

Comment:
Should there be a static assert that `T` is nothrow default constructible?

Reply:
I think the fact that it compiles is equivalent to asserting this.